<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Yi Nengjun" />

<meta name="date" content="2019-02-19" />

<title>Introduction to BhGLM</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>

</head>

<body>




<h1 class="title toc-ignore">Introduction to BhGLM</h1>
<h4 class="author"><em>Yi Nengjun</em></h4>
<h4 class="date"><em>2019-02-19</em></h4>



<p>This package provides functions for setting up and fitting various Bayesian hierarchical models (generalized linear models (GLMs), Cox survival models, negative binomial models, and ordered logistic or probit regressions), for numerically and graphically summarizing the fitted models, and for evaluating the predictive performance. Several types of priors on the coefficients can be used: double-exponential, Student-t, mixture double-exponential, and mixture t. The models are fitted by using fast algorithms for estimating posterior modes rather than MCMC. The methods can be used to analyze not only general data but also large-scale genomic data (i.e., detecting disease-associated genes or variants, predictive and prognostic modeling of diseases and traits).</p>
<div id="data-preparation-functions" class="section level2">
<h2>Data Preparation Functions</h2>
<ul>
<li><p><code>covariates</code>: This function standardizes continuous variables, transforms categorical variables to indicator variables and centers the transformed variables.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># fake data</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">age =<span class="st"> </span><span class="kw">rnorm</span>(<span class="dv">50</span>, <span class="dv">30</span>, <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">sex =<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x =</span> <span class="kw">c</span>(<span class="st">&quot;male&quot;</span>, <span class="st">&quot;female&quot;</span>), <span class="dt">size =</span> <span class="dv">50</span>, <span class="dt">replace =</span> T, <span class="dt">prob =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">diet =<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">5</span>), <span class="dt">size =</span> <span class="dv">50</span>, <span class="dt">replace =</span> T, <span class="dt">prob =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.7</span>))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">race =<span class="st"> </span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">&quot;Asian&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Black&quot;</span>), <span class="dt">size =</span> <span class="dv">50</span>, <span class="dt">replace =</span> T, <span class="dt">prob =</span> <span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">0.4</span>, <span class="fl">0.4</span>))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">x.con =<span class="st"> </span>age <span class="co"># continuous variable</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8">x.cat =<span class="st"> </span><span class="kw">cbind</span>(sex, diet, race) <span class="co"># categorical variables</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"></a>
<a class="sourceLine" id="cb1-10" data-line-number="10">x.con[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>] =<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11">x.cat[<span class="dv">1</span>,] =<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13">x1 =<span class="st"> </span><span class="kw">covariates</span>(<span class="dt">x.con =</span> x.con, <span class="dt">x.cat =</span> x.cat, <span class="dt">con.rescale =</span> F, <span class="dt">cat.center =</span> F, </a>
<a class="sourceLine" id="cb1-14" data-line-number="14">                <span class="dt">fill.missing =</span> T)</a>
<a class="sourceLine" id="cb1-15" data-line-number="15">x1</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">x2 =<span class="st"> </span><span class="kw">covariates</span>(<span class="dt">x.con =</span> x.con, <span class="dt">x.cat =</span> x.cat, <span class="dt">con.rescale =</span> T, <span class="dt">cat.center =</span> F, </a>
<a class="sourceLine" id="cb1-17" data-line-number="17">                <span class="dt">fill.missing =</span> T)</a>
<a class="sourceLine" id="cb1-18" data-line-number="18">x2</a>
<a class="sourceLine" id="cb1-19" data-line-number="19">x3 =<span class="st"> </span><span class="kw">covariates</span>(<span class="dt">x.con =</span> x.con, <span class="dt">x.cat =</span> x.cat, <span class="dt">con.rescale =</span> T, <span class="dt">cat.center =</span> T, </a>
<a class="sourceLine" id="cb1-20" data-line-number="20">                <span class="dt">fill.missing =</span> T)</a>
<a class="sourceLine" id="cb1-21" data-line-number="21">x3</a></code></pre></div></li>
<li><p><code>make.inter</code>: This function is to construct a design matrix of interactions from two main-effect design matrices.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">data</span>(fake.cv) <span class="co">#load data from BhGLM</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">cov1 =<span class="st"> </span>fake.cv[, <span class="dv">3</span>, drop =<span class="st"> </span>F] <span class="co">#get the first covariate</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">geno =<span class="st"> </span>fake.cv[, <span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>)] <span class="co">#get genotype data</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">x.main =<span class="st"> </span><span class="kw">make.main</span>(<span class="dt">geno =</span> geno, <span class="dt">model =</span> <span class="st">&quot;Cockerham&quot;</span>, <span class="dt">fill.missing =</span> T)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">x.GxE =<span class="st"> </span><span class="kw">make.inter</span>(cov1, x.main, <span class="dt">back =</span> <span class="dv">0</span>) <span class="co">#interactions between cov1 and all main effects</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">x.GxG1 =<span class="st"> </span><span class="kw">make.inter</span>(x.main[, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>], x.main[, <span class="dv">11</span><span class="op">:</span><span class="dv">20</span>], <span class="dt">back =</span> <span class="dv">1</span>) <span class="co"># some GxG interactions </span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">x.GxG2 =<span class="st"> </span><span class="kw">make.inter</span>(x.main, x.main, <span class="dt">back =</span> <span class="dv">1</span>) <span class="co"># all GxG interactions </span></a></code></pre></div></li>
<li><p><code>make.main</code>: This function is to construct a design matrix of main effects from genotypic data of genetic markers. The genotypic data can include missing values. Different genetic models can be used, which transform the three-level (or two-level for a backcross) genotypic data to main-effect predictors.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">## Example 1: genetic association studies</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">data</span>(fake.cv) <span class="co">#load data from BhGLM</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">geno =<span class="st"> </span>fake.cv[, <span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>)] <span class="co">#get genotype data</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">x =<span class="st"> </span><span class="kw">make.main</span>(<span class="dt">geno =</span> geno, <span class="dt">model =</span> <span class="st">&quot;additive&quot;</span>, <span class="dt">fill.missing =</span> T)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">x =<span class="st"> </span><span class="kw">make.main</span>(<span class="dt">geno =</span> geno, <span class="dt">model =</span> <span class="st">&quot;Cockerham&quot;</span>, <span class="dt">fill.missing =</span> T, <span class="dt">ind.group =</span> fake.cv<span class="op">$</span>y1)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">## Example 2: QTL mapping in experimental crosses</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="kw">require</span>(qtl)</a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="kw">require</span>(qtlbim)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="kw">data</span>(listeria)</a>
<a class="sourceLine" id="cb3-11" data-line-number="11">listeria</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">x =<span class="st"> </span><span class="kw">make.main</span>(<span class="dt">geno =</span> listeria, <span class="dt">model =</span> <span class="st">&quot;Cockerham&quot;</span>, <span class="dt">fill.missing =</span> T, <span class="dt">geno.order =</span> F, <span class="dt">loci.names =</span> <span class="st">&quot;position&quot;</span>)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">## then we can follow the example in bglm to detect QTL</span></a></code></pre></div></li>
</ul>
</div>
<div id="model-functions" class="section level2">
<h2>Model Functions</h2>
<ul>
<li><p><code>bcoxph</code> This function is to set up Bayesian hierarchical Cox proportonal hazards model and to fit the model using the EM Newton-Raphson algorithm. As in bglm, several types of priors on the coefficients can be used. The Bayesian hierarchical Cox model includes classical Cox model and ridge Cox regression as special cases. It can be used for analyzing general survival data and large-scale and highly-correlated variables (for example, detecting disease-associated factors and predicting survival times).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">library</span>(BhGLM)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">library</span>(survival)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">N =<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5">K =<span class="st"> </span><span class="dv">100</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6">x =<span class="st"> </span><span class="kw">sim.x</span>(<span class="dt">n=</span>N, <span class="dt">m=</span>K, <span class="dt">corr=</span><span class="fl">0.6</span>) <span class="co"># simulate correlated continuous variables  </span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7">h =<span class="st"> </span><span class="kw">rep</span>(<span class="fl">0.1</span>, <span class="dv">4</span>) <span class="co"># assign four non-zero main effects to have the assumed heritabilty </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8">nz =<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">seq</span>(<span class="dv">5</span>, K, <span class="dt">by=</span>K<span class="op">/</span><span class="kw">length</span>(h))); nz</a>
<a class="sourceLine" id="cb4-9" data-line-number="9">yy =<span class="st"> </span><span class="kw">sim.y</span>(<span class="dt">x=</span>x[, nz], <span class="dt">mu=</span><span class="dv">0</span>, <span class="dt">herit=</span>h, <span class="dt">p.neg=</span><span class="fl">0.5</span>) <span class="co"># simulate responses</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10">yy<span class="op">$</span>coefs</a>
<a class="sourceLine" id="cb4-11" data-line-number="11"></a>
<a class="sourceLine" id="cb4-12" data-line-number="12">y =<span class="st"> </span>yy<span class="op">$</span>y.surv</a>
<a class="sourceLine" id="cb4-13" data-line-number="13">d =<span class="st"> </span><span class="kw">table</span>(y[,<span class="dv">2</span>]); d[<span class="dv">1</span>]<span class="op">/</span><span class="kw">sum</span>(d) <span class="co"># cencoring proportion</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co"># jointly update</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co"># Compare with conventional and ridge Cox model</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="dt">cex.axis =</span> <span class="dv">1</span>, <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>))</a>
<a class="sourceLine" id="cb4-19" data-line-number="19">gap =<span class="st"> </span><span class="dv">10</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"></a>
<a class="sourceLine" id="cb4-21" data-line-number="21">ps =<span class="st"> </span><span class="fl">0.05</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22">f1 =<span class="st"> </span><span class="kw">bcoxph</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">prior =</span> <span class="st">&quot;de&quot;</span>, <span class="dt">prior.scale =</span> ps)</a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co"># summary.bh(f1)</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="kw">plot.bh</span>(f1, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;Cox with double-exponential&quot;</span>)</a>
<a class="sourceLine" id="cb4-25" data-line-number="25"></a>
<a class="sourceLine" id="cb4-26" data-line-number="26">f2 =<span class="st"> </span><span class="kw">bcoxph</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">prior =</span> <span class="st">&quot;t&quot;</span>, <span class="dt">prior.scale =</span> ps<span class="op">/</span><span class="fl">1.4</span>)</a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co"># summary.bh(f2)</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="kw">plot.bh</span>(f2, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;Cox with t&quot;</span>)</a>
<a class="sourceLine" id="cb4-29" data-line-number="29"></a>
<a class="sourceLine" id="cb4-30" data-line-number="30">ss =<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.04</span>, <span class="fl">0.5</span>) </a>
<a class="sourceLine" id="cb4-31" data-line-number="31">f3 =<span class="st"> </span><span class="kw">bcoxph</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">prior =</span> <span class="st">&quot;mde&quot;</span>, <span class="dt">ss =</span> ss)</a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co"># summary.bh(f3)</span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="kw">plot.bh</span>(f3, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;Cox with mixture double exponential&quot;</span>) </a>
<a class="sourceLine" id="cb4-34" data-line-number="34"></a>
<a class="sourceLine" id="cb4-35" data-line-number="35">ss =<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.04</span>, <span class="fl">0.5</span>) </a>
<a class="sourceLine" id="cb4-36" data-line-number="36">f4 =<span class="st"> </span><span class="kw">bcoxph</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">prior =</span> <span class="st">&quot;mt&quot;</span>, <span class="dt">ss =</span> ss)</a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co"># summary.bh(f4)</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"><span class="kw">plot.bh</span>(f4, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;Cox with mixture t&quot;</span>) </a>
<a class="sourceLine" id="cb4-39" data-line-number="39"></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="co"># group-wise update</span></a>
<a class="sourceLine" id="cb4-42" data-line-number="42">ps =<span class="st"> </span><span class="fl">0.05</span></a>
<a class="sourceLine" id="cb4-43" data-line-number="43">f1 =<span class="st"> </span><span class="kw">bcoxph</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">prior =</span> <span class="st">&quot;de&quot;</span>, <span class="dt">method.coef =</span> <span class="dv">50</span>, <span class="dt">prior.scale =</span> ps)</a>
<a class="sourceLine" id="cb4-44" data-line-number="44"><span class="co">#summary.bh(f1)</span></a>
<a class="sourceLine" id="cb4-45" data-line-number="45"><span class="kw">plot.bh</span>(<span class="dt">coefs =</span> f2<span class="op">$</span>coefficients, <span class="dt">threshold =</span> <span class="dv">10</span>, <span class="dt">gap =</span> gap)  </a></code></pre></div></li>
<li><p><code>bglm</code>: Bayesian Hierarchical Generalized Linear Models (GLMs)</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">N =<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">K =<span class="st"> </span><span class="dv">100</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3">x =<span class="st"> </span><span class="kw">sim.x</span>(<span class="dt">n=</span>N, <span class="dt">m=</span>K, <span class="dt">corr=</span><span class="fl">0.6</span>) <span class="co"># simulate correlated continuous variables  </span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4">h =<span class="st"> </span><span class="kw">rep</span>(<span class="fl">0.1</span>, <span class="dv">4</span>) <span class="co"># assign four non-zero main effects to have the assumed heritabilty </span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5">nz =<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">seq</span>(<span class="dv">5</span>, K, <span class="dt">by=</span>K<span class="op">/</span><span class="kw">length</span>(h))); nz</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">yy =<span class="st"> </span><span class="kw">sim.y</span>(<span class="dt">x=</span>x[, nz], <span class="dt">mu=</span><span class="dv">0</span>, <span class="dt">herit=</span>h, <span class="dt">p.neg=</span><span class="fl">0.5</span>, <span class="dt">sigma=</span><span class="fl">1.6</span>, <span class="dt">theta=</span><span class="dv">2</span>) <span class="co"># simulate responses</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7">yy<span class="op">$</span>coefs</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co"># y = yy$y.normal; fam = gaussian; y = scale(y)</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co"># y = yy$y.ordinal; fam = binomial</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12">y =<span class="st"> </span>yy<span class="op">$</span>y.nb; fam =<span class="st"> </span>NegBin</a>
<a class="sourceLine" id="cb5-13" data-line-number="13"></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co"># jointly fit all variables (can be slow if m is large)</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="dt">cex.axis =</span> <span class="dv">1</span>, <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>))</a>
<a class="sourceLine" id="cb5-16" data-line-number="16">gap =<span class="st"> </span><span class="dv">10</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"></a>
<a class="sourceLine" id="cb5-18" data-line-number="18">ps =<span class="st"> </span><span class="fl">0.05</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19">f1 =<span class="st"> </span><span class="kw">bglm</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">family =</span> fam, <span class="dt">prior =</span> <span class="st">&quot;de&quot;</span>, <span class="dt">prior.scale =</span> ps)   </a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="kw">plot.bh</span>(f1, <span class="dt">vars.rm =</span> <span class="dv">1</span>, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;de&quot;</span>)  </a>
<a class="sourceLine" id="cb5-21" data-line-number="21"></a>
<a class="sourceLine" id="cb5-22" data-line-number="22">f2 =<span class="st"> </span><span class="kw">bglm</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">family =</span> fam, <span class="dt">prior =</span> <span class="st">&quot;t&quot;</span>, <span class="dt">prior.scale =</span> ps<span class="op">/</span><span class="fl">1.4</span>)   </a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="kw">plot.bh</span>(f2, <span class="dt">vars.rm =</span> <span class="dv">1</span>, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;t&quot;</span>)  </a>
<a class="sourceLine" id="cb5-24" data-line-number="24"></a>
<a class="sourceLine" id="cb5-25" data-line-number="25">ss =<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.04</span>, <span class="fl">0.5</span>) </a>
<a class="sourceLine" id="cb5-26" data-line-number="26">f3 =<span class="st"> </span><span class="kw">bglm</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">family =</span> fam, <span class="dt">prior =</span> <span class="st">&quot;mde&quot;</span>, <span class="dt">ss =</span> ss)   </a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="kw">plot.bh</span>(f3, <span class="dt">vars.rm =</span> <span class="dv">1</span>, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;mde&quot;</span>)  </a>
<a class="sourceLine" id="cb5-28" data-line-number="28"></a>
<a class="sourceLine" id="cb5-29" data-line-number="29">ss =<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.04</span>, <span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb5-30" data-line-number="30">f4 =<span class="st"> </span><span class="kw">bglm</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">family =</span> fam, <span class="dt">prior =</span> <span class="st">&quot;mt&quot;</span>, <span class="dt">ss =</span> ss, <span class="dt">prior.df=</span><span class="ot">Inf</span>)   </a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="kw">plot.bh</span>(f4, <span class="dt">vars.rm =</span> <span class="dv">1</span>, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;mt&quot;</span>)  </a>
<a class="sourceLine" id="cb5-32" data-line-number="32"></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"></a>
<a class="sourceLine" id="cb5-34" data-line-number="34"><span class="co"># group-wise update (can be much faster if m is large)</span></a>
<a class="sourceLine" id="cb5-35" data-line-number="35">ps =<span class="st"> </span><span class="fl">0.05</span></a>
<a class="sourceLine" id="cb5-36" data-line-number="36">f1 =<span class="st"> </span><span class="kw">bglm</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">family =</span> fam, <span class="dt">prior =</span> <span class="st">&quot;de&quot;</span>, <span class="dt">method.coef =</span> <span class="dv">50</span>, <span class="dt">prior.scale =</span> ps) <span class="co"># update 50 coefficients at a time</span></a>
<a class="sourceLine" id="cb5-37" data-line-number="37"><span class="kw">plot.bh</span>(f1, <span class="dt">vars.rm =</span> <span class="dv">1</span>, <span class="dt">threshold =</span> <span class="fl">0.01</span>, <span class="dt">gap =</span> gap)  </a></code></pre></div></li>
<li><p><code>bmlasso</code>: Bayesian Spike-and-Slab Lasso or Ridge GLMs and Cox Models</p></li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">library</span>(BhGLM)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">library</span>(survival)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw">library</span>(glmnet)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"></a>
<a class="sourceLine" id="cb6-6" data-line-number="6">N =<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7">K =<span class="st"> </span><span class="dv">100</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8">x =<span class="st"> </span><span class="kw">sim.x</span>(<span class="dt">n=</span>N, <span class="dt">m=</span>K, <span class="dt">corr=</span><span class="fl">0.6</span>) <span class="co"># simulate correlated continuous variables  </span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9">h =<span class="st"> </span><span class="kw">rep</span>(<span class="fl">0.1</span>, <span class="dv">4</span>) <span class="co"># assign four non-zero main effects to have the assumed heritabilty </span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10">nz =<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">seq</span>(<span class="dv">5</span>, K, <span class="dt">by=</span>K<span class="op">/</span><span class="kw">length</span>(h))); nz</a>
<a class="sourceLine" id="cb6-11" data-line-number="11">yy =<span class="st"> </span><span class="kw">sim.y</span>(<span class="dt">x=</span>x[, nz], <span class="dt">mu=</span><span class="dv">0</span>, <span class="dt">herit=</span>h, <span class="dt">p.neg=</span><span class="fl">0.5</span>, <span class="dt">sigma=</span><span class="fl">1.6</span>) <span class="co"># simulate responses</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12">yy<span class="op">$</span>coefs</a>
<a class="sourceLine" id="cb6-13" data-line-number="13"></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co"># y = yy$y.normal; fam = &quot;gaussian&quot;; y = scale(y)</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co"># y = yy$y.ordinal; fam = &quot;binomial&quot;</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16">y =<span class="st"> </span>yy<span class="op">$</span>y.surv; fam =<span class="st"> &quot;cox&quot;</span> </a>
<a class="sourceLine" id="cb6-17" data-line-number="17"></a>
<a class="sourceLine" id="cb6-18" data-line-number="18">group =<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#group = rep(0, 21)</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#for(j in 1:length(group)) group[j] = (j-1) * K/(length(group)-1)</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co"># lasso and mixture lasso</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"></a>
<a class="sourceLine" id="cb6-24" data-line-number="24">f1 =<span class="st"> </span><span class="kw">glmNet</span>(x, y, <span class="dt">family =</span> fam, <span class="dt">ncv =</span> <span class="dv">1</span>) </a>
<a class="sourceLine" id="cb6-25" data-line-number="25"></a>
<a class="sourceLine" id="cb6-26" data-line-number="26">ps =<span class="st"> </span>f1<span class="op">$</span>prior.scale; ps </a>
<a class="sourceLine" id="cb6-27" data-line-number="27">ss =<span class="st"> </span><span class="kw">c</span>(ps, <span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb6-28" data-line-number="28">f2 =<span class="st"> </span><span class="kw">bmlasso</span>(x, y, <span class="dt">family =</span> fam, <span class="dt">prior =</span> <span class="st">&quot;mde&quot;</span>, <span class="dt">ss =</span> ss, <span class="dt">group =</span> group)</a>
<a class="sourceLine" id="cb6-29" data-line-number="29"></a>
<a class="sourceLine" id="cb6-30" data-line-number="30"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>))</a>
<a class="sourceLine" id="cb6-31" data-line-number="31">gap =<span class="st"> </span><span class="dv">10</span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="kw">plot.bh</span>(<span class="dt">coefs =</span> f1<span class="op">$</span>coef, <span class="dt">threshold =</span> f1<span class="op">$</span>df, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;lasso&quot;</span>) </a>
<a class="sourceLine" id="cb6-33" data-line-number="33"><span class="kw">plot.bh</span>(<span class="dt">coefs =</span> f2<span class="op">$</span>coef, <span class="dt">threshold =</span> f2<span class="op">$</span>df, <span class="dt">gap =</span> gap, <span class="dt">main =</span> <span class="st">&quot;mixture lasso&quot;</span>) </a></code></pre></div>
<ul>
<li><code>bpolr</code>: Bayesian Hierarchical Ordered Logistic or Probit Regressions for Ordinal Response</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">library</span>(BhGLM)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"></a>
<a class="sourceLine" id="cb7-4" data-line-number="4">N =<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5">K =<span class="st"> </span><span class="dv">100</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6">x =<span class="st"> </span><span class="kw">sim.x</span>(<span class="dt">n =</span> N, <span class="dt">m =</span> K, <span class="dt">corr =</span> <span class="fl">0.6</span>) <span class="co"># simulate correlated variables  </span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7">x =<span class="st"> </span><span class="kw">as.matrix</span>(x)</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">h =<span class="st"> </span><span class="kw">rep</span>(<span class="fl">0.1</span>, <span class="dv">4</span>) <span class="co"># assign non-zero effects to have the assumed heritabilty </span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9">nz =<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">seq</span>(<span class="dv">5</span>, K, <span class="dt">by=</span>K<span class="op">/</span><span class="kw">length</span>(h))); nz</a>
<a class="sourceLine" id="cb7-10" data-line-number="10">yy =<span class="st"> </span><span class="kw">sim.y</span>(<span class="dt">x=</span>x[, nz], <span class="dt">mu =</span> <span class="dv">10</span>, <span class="dt">herit=</span>h, <span class="dt">p.neg=</span><span class="fl">0.5</span>, <span class="dt">sigma=</span><span class="fl">1.6</span>, <span class="dt">quantiles =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.6</span>)) <span class="co"># simulate responses</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11">yy<span class="op">$</span>coefs</a>
<a class="sourceLine" id="cb7-12" data-line-number="12">y =<span class="st"> </span><span class="kw">as.factor</span>(yy<span class="op">$</span>y.ordinal)</a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="kw">table</span>(y)</a>
<a class="sourceLine" id="cb7-14" data-line-number="14"></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="dt">cex.axis =</span> <span class="dv">1</span>, <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>))</a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co"># classical ordered logistic regression</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="kw">library</span>(MASS)</a>
<a class="sourceLine" id="cb7-19" data-line-number="19">f1 =<span class="st"> </span><span class="kw">polr</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">Hess =</span> T)</a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="kw">plot.bh</span>(f1, <span class="dt">gap =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="kw">summary.bh</span>(f1)</a>
<a class="sourceLine" id="cb7-22" data-line-number="22"></a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="co"># equivalent to classical ordered logistic regression</span></a>
<a class="sourceLine" id="cb7-24" data-line-number="24">f2 =<span class="st"> </span><span class="kw">bpolr</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">prior.scale =</span> <span class="ot">Inf</span>)</a>
<a class="sourceLine" id="cb7-25" data-line-number="25"><span class="kw">plot.bh</span>(f2, <span class="dt">gap =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb7-26" data-line-number="26"><span class="kw">summary.bh</span>(f2)</a>
<a class="sourceLine" id="cb7-27" data-line-number="27"></a>
<a class="sourceLine" id="cb7-28" data-line-number="28"><span class="co"># hierarchical ordered logistic regression</span></a>
<a class="sourceLine" id="cb7-29" data-line-number="29">f3 =<span class="st"> </span><span class="kw">bpolr</span>(y <span class="op">~</span><span class="st"> </span>., <span class="dt">data =</span> x, <span class="dt">prior.scale =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb7-30" data-line-number="30"><span class="kw">plot.bh</span>(f3, <span class="dt">gap =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb7-31" data-line-number="31"><span class="kw">summary.bh</span>(f3)</a></code></pre></div>
<ul>
<li><code>glmNet</code>: Fit a GLM or Cox Model with lasso or elastic net</li>
</ul>
</div>
<div id="summary-and-validation-functions" class="section level2">
<h2>Summary And Validation Functions</h2>
<ul>
<li><p><code>cv.bh</code>: Cross-Validation for Bayesian Hierarchical GLMs and Cox Model or Elastic Net (including Lasso and Ridge)</p></li>
<li><p><code>df.adj</code>: df.adj Calculating Adjusted Degree of Freedom (Effective Number of Parameters) in Bayesian Hierarchical GLMs</p></li>
<li>Functions for summarizing a fitted Cox survival model or cross-validation
<ul>
<li><code>surv.curves</code></li>
<li><code>Cindex</code></li>
<li><code>aucCox</code></li>
<li><code>peCox</code></li>
</ul></li>
<li><p><code>mc.adj</code>: Adjust P-values for Multiple Comparisons</p></li>
<li><code>predict.bh</code>: Predictions for a Bayesian hierarchical model</li>
<li><p><code>summary.bh</code>: Numerically Summarizing Bayesian Hierarchical Model Fits</p></li>
</ul>
</div>
<div id="utility-functions" class="section level2">
<h2>Utility Functions</h2>
<ul>
<li><p><code>plot.bh</code>: Graphically Summarizing Bayesian Hierarchical Model Fits</p></li>
<li><p><code>geno.freq</code>: Summarizing Genotypic Data</p></li>
<li><p><code>simulation</code>: Simulating Predictor Variables and Response Data and Summarizing the Analysis of Simulated Data</p></li>
</ul>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
